### PR TITLE
Make Viewport::screen_to_map_pos() return coords

### PR DESCRIPTION
**Pull request by tupaschoal**
As proposed by @duncanspumpkin on #10131

> This is horrible. We should refactor this.
>
> mapX and mapY are CoordsXY values but you need to set them to screenCoords.x and screenCoords.y if clickedElement is nullptr below as screen_pos_to_map_pos is a horrible function that takes in screen coords and spits out coordsxy to the same variable.

Notes to future:
- Make `screen_get_map_xy_side` and `screen_get_map_xy` return `CoordsXY`.
- Make `screen_pos_to_map_pos`, `screen_get_map_xy_side` receive `ScreenCoordsXY`

### CODE REVIEW

File: src/openrct2/interface/Viewport.cpp [hunk 1]
```diff
@@ -997,23 +997,27 @@ static void viewport_paint_weather_gloom(rct_drawpixelinfo* dpi)
...
-    screen_get_map_xy(*x, *y, x, y, nullptr);
-    if (*x == LOCATION_NULL)
-        return;
+    int16_t mapX = 0;
+    int16_t mapY = 0;
+    screen_get_map_xy(x, y, &mapX, &mapY, nullptr);
+    if (mapX == LOCATION_NULL)
+        return {};
+
+    CoordsXY mapCoords = {mapX, mapY};
```
Comment (tupaschoal) [inline src/openrct2/interface/Viewport.cpp]:
```suggestion
    CoordsXY mapCoords = { mapX, mapY };
```

File: src/openrct2-ui/windows/TileInspector.cpp [hunk 2]
```diff
@@ -1282,15 +1284,16 @@ static void window_tile_inspector_update_selected_tile(rct_window* w, int32_t x,
...
             return;
         }
 
         // Tile is already selected
-        if (windowTileInspectorTileSelected && mapX == windowTileInspectorToolMap.x && mapY == windowTileInspectorToolMap.y)
+        if (windowTileInspectorTileSelected && mapCoords.x == windowTileInspectorToolMap.x
+            && mapCoords.y == windowTileInspectorToolMap.y)
         {
             return;
         }
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/TileInspector.cpp]: mapX and mapY are used below and they would have the wrong value

Comment (tupaschoal) [inline src/openrct2-ui/windows/TileInspector.cpp]:
```suggestion
         }
    mapX = mapCoords.x;
    mapY = mapCoords.y;
```

File: src/openrct2-ui/windows/TileInspector.cpp [hunk 3]
```diff
@@ -1282,18 +1284,21 @@ static void window_tile_inspector_update_selected_tile(rct_window* w, int32_t x,
...
 
         // Tile is already selected
-        if (windowTileInspectorTileSelected && mapX == windowTileInspectorToolMap.x && mapY == windowTileInspectorToolMap.y)
+        if (windowTileInspectorTileSelected && mapCoords.x == windowTileInspectorToolMap.x
+            && mapCoords.y == windowTileInspectorToolMap.y)
         {
             return;
-        }
+         }
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TileInspector.cpp]:
```suggestion
        }
```

File: src/openrct2-ui/windows/TileInspector.cpp [hunk 4]
```diff
@@ -1282,18 +1284,21 @@ static void window_tile_inspector_update_selected_tile(rct_window* w, int32_t x,
...
         }
 
         // Tile is already selected
-        if (windowTileInspectorTileSelected && mapX == windowTileInspectorToolMap.x && mapY == windowTileInspectorToolMap.y)
+        if (windowTileInspectorTileSelected && mapCoords.x == windowTileInspectorToolMap.x
+            && mapCoords.y == windowTileInspectorToolMap.y)
         {
             return;
         }
+    mapX = mapCoords.x;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TileInspector.cpp]:
```suggestion
        mapX = mapCoords.x;
```

File: src/openrct2-ui/windows/TileInspector.cpp [hunk 5]
```diff
@@ -1282,18 +1284,21 @@ static void window_tile_inspector_update_selected_tile(rct_window* w, int32_t x,
...
+    mapY = mapCoords.y;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TileInspector.cpp]:
```suggestion
        mapY = mapCoords.y;
```

File: src/openrct2/interface/Viewport.cpp [hunk 6]
```diff
@@ -995,23 +995,27 @@ static void viewport_paint_weather_gloom(rct_drawpixelinfo* dpi)
...
+CoordsXY screen_pos_to_map_pos(int16_t x, int16_t y, int32_t* direction)
 {
-    screen_get_map_xy(*x, *y, x, y, nullptr);
-    if (*x == LOCATION_NULL)
-        return;
+    int16_t mapX = 0;
+    int16_t mapY = 0;
+    screen_get_map_xy(x, y, &mapX, &mapY, nullptr);
+    if (mapX == LOCATION_NULL)
+        return {};
```
Comment (duncanspumpkin) [inline src/openrct2/interface/Viewport.cpp]: Doesn't return the error upwards. Causes #10410

**Pull request merged**